File: apps/lesson-plan-generator/internal/pipeline/assembler.go
package pipeline

import (
	"math"

	"lesson-plan-generator/internal/domain"
)

// step 7 – assemble the final VideoPlan JSON contract.
func assemble(lessonID string, segments []domain.AudioSegment, fullAudioFile string, scenes []map[string]any) domain.VideoPlan {
	var total float64
	for _, s := range segments {
		total += s.DurationSec
	}

	return domain.VideoPlan{
		LessonID: lessonID,
		AudioTrack: domain.AudioTrack{
			Segments:      segments,
			FullAudioFile: fullAudioFile,
			TotalDuration: math.Round(total*1000) / 1000,
		},
		Scenes: scenes,
	}
}

File: apps/lesson-plan-generator/app/steps/assembler.py
"""Step 7 – Assemble the final VideoPlan JSON contract."""

from app.models import AudioSegment, AudioTrack, VideoPlan


def assemble(
    lesson_id: str,
    audio_segments: list[AudioSegment],
    full_audio_file: str,
    scenes: list[dict],
) -> VideoPlan:
    total_duration = round(sum(s.duration_sec for s in audio_segments), 3)

    audio_track = AudioTrack(
        segments=audio_segments,
        full_audio_file=full_audio_file,
        total_duration=total_duration,
    )

    return VideoPlan(
        lesson_id=lesson_id,
        audio_track=audio_track,
        scenes=scenes,
    )
